code/get_analysis.py

```diff
@@ -0,0 +1,81 @@
+        bot.send_message(chat_id, "No category budget available", reply_markup=types.ReplyKeyboardRemove())
+    category_budget = {}
+    for cat in helper.spend_categories:
+        if helper.isCategoryBudgetByCategoryAvailable(chat_id, cat):
+            category_budget[cat] = helper.getCategoryBudgetByCategory(chat_id, cat)
+    
+    _, ax = plt.subplots()
+    ax.pie(category_budget.values(), labels=category_budget.keys(), autopct='%1.1f%%')
+    random_time = time.time()
+    img_name = str(random_time)+".png"
+    plt.savefig(img_name)
+    bot.send_photo(chat_id, photo=open(img_name, 'rb'), reply_markup=types.ReplyKeyboardRemove())
+
+def viewSpendWise(chat_id, bot):
+    category_spend = {}
+    for cat in helper.spend_categories:
+        spend = helper.calculate_total_spendings_for_cateogory_chat_id(chat_id,cat)
+        if spend != 0:
+            category_spend[cat] = spend
+
+    if category_spend.keys() == None:
+        bot.send_message(chat_id, "No category spend available", reply_markup=types.ReplyKeyboardRemove())
+    
+    _, ax = plt.subplots()
+    ax.pie(category_spend.values(), labels=category_spend.keys(), autopct='%1.1f%%')
+    random_time = time.time()
+    img_name = str(random_time)+".png"
+    plt.savefig(img_name)
+    bot.send_photo(chat_id, photo=open(img_name, 'rb'), reply_markup=types.ReplyKeyboardRemove())
+
+def viewRemaining(chat_id, bot):
+    if not helper.isCategoryBudgetAvailable(chat_id):
+        bot.send_message(chat_id, "No category budget available", reply_markup=types.ReplyKeyboardRemove())
+
+    category_spend_percent = {}
+    for cat in helper.spend_categories:
+        if helper.isCategoryBudgetByCategoryAvailable(chat_id, cat):
+            percent = helper.calculateRemainingCateogryBudgetPercent(chat_id, cat)
+            category_spend_percent[cat] = percent
+    
+    labels = tuple(category_spend_percent.keys())
+    print(labels)
+    type(labels)
+
+    remaining_val_list = [100 - x for x in list(category_spend_percent.values())]
+
+    weight_counts = {
+        "Used": list(category_spend_percent.values()),
+        "Remaining": remaining_val_list,
+    }
+    print(weight_counts)
+
+    width = 0.5
+
+    _, ax = plt.subplots()
+    # simply impossible to submit anything other than an np ndarray here
+    # matplotlib simply fails to recognize shape of any other "array-like"
+    #object
+    bottom = np.zeros(len(list(category_spend_percent.values())))
+
+    for boolean, weight_count in weight_counts.items():
+        print(boolean, weight_count)
+        ax.bar(labels, weight_count, width, label=boolean, bottom=bottom)
+        bottom += weight_count
+
+    ax.set_title("Category-wise budget consumed")
+    ax.legend(loc="upper right")
+
+    random_time = time.time()
+    img_name = str(random_time)+".png"
+    plt.savefig(img_name)
+    bot.send_photo(chat_id, photo=open(img_name, 'rb'), reply_markup=types.ReplyKeyboardRemove())
```

Comment:
While percentages are a good indicator, as a user, I would also like to know exactly how much money I have spent or have left to spend in my budget without having to calculate based on percentages.

Reply:
Will add as enhancement later. Requires lot of refactor.